### ISSUE TITLE
Confidence parameter

### ISSUE THREAD

**Issue by Clayes97** [mkazhdan/PoissonRecon#274]
Hello, I am not understanding what the confidence parameter represents.

Should It be something related on how much I want to fulfill the constraint related to the screeming? Is therefore what is called alpha in the papers? 

Thanks a lot for your answer

**Comment by mkazhdan** [comment]
Not quite. The alpha parameter for screening is defined by --pointWeight.

In addition, it is possible that you have more confidence in some samples than others. (For example, if a point is facing the camera you are likely to be more confident than in a point viewed at a grazing angle).
To support this, the code allows you to encode your confidence in the magnitude of the normal.
The code then computes the magnitude of the normal raises it to the power of --confidence, and further scales the screening weight _at that sample_ by the exponentiated normal magnitude. (By default --confidence has value 0 so that the local weight is 1 for all samples, but you could set it to a positive value if you have encoded confidence in the magnitude of a sample's normal.)

**Comment by Clayes97** [comment]
So, if I got it: There are some points  ```p ``` of my point cloud for which I'm suspicious about their position (due to the sensor). Therefore, I weight their contribution in the screening constraint as ```w (p) = ||n_p||^{(confidence)}``` in order to say that for those points I dont want a perfect interpolation?

**Comment by mkazhdan** [comment]
Yes. Assuming that when you generated the point cloud you made the magnitudes of the normals of those points smaller.

**Comment by mkazhdan** [comment]
If the points have normals with magnitude 1, the associated weight will be 1 regardless of the confidence. So as long as you can scale the unit normals by the samples confidence (in the range [0,1])...
